src/pymovements/gaze/_utils/parsing_begaze.py

```diff
@@ -0,0 +1,557 @@
+# Copyright (c) 2025 The pymovements Project Authors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""BeGaze parsing module (internal)."""
+from __future__ import annotations
+
+__all__ = [
+    'parse_begaze',
+]
+
+import datetime
+import re
+from collections import defaultdict
+from pathlib import Path
+from typing import Any
+
+import warnings
+
+import numpy as np
+import polars as pl
+
+from pymovements.gaze._utils.parsing import compile_patterns, get_pattern_keys, \
+    check_nan, _calculate_data_loss_ratio
+
+# Regular expressions for BeGaze header metadata lines with named groups.
+# Note: we compile regexes for performance and to support minor whitespace variations.
+BEGAZE_META_REGEXES: list[re.Pattern[str]] = [
+    re.compile(r'^##\s+Date:\s+(?P<date>.+?)\s*$'),
+    re.compile(r'^##\s+Sample\s+Rate:\s+(?P<sampling_rate>\d+(?:[.]\d+)?)\s*$'),
+]
+
+
+def _parse_begaze_meta_line(line: str) -> dict[str, Any]:
+    """Parse a single BeGaze '##' metadata line using predefined regexes.
+
+    Returns an empty dict when the line does not match any known pattern.
+    Performs light casting for known fields (e.g., float for sampling rate, datetime for date).
+    """
+    for regex in BEGAZE_META_REGEXES:
+        if match := regex.match(line):
+            groupdict = match.groupdict()
+            # Casting and processing for known fields
+            if 'sampling_rate' in groupdict and groupdict['sampling_rate'] is not None:
```

Comment:
Can we simplify this to:
```suggestion
            if groupdict.get('sampling_rate') is not None:
```